From a31a5d1a2ac67bbdcd676ee2ac738741a3cd53f6 Mon Sep 17 00:00:00 2001
From: Niranjan H Y <niranjan.hy@ti.com>
Date: Fri, 13 Mar 2026 00:18:28 +0530
Subject: [PATCH 20/27] dt-bindings: sound: Add bindings for TI TAC5x1x codec

Add device tree bindings for the Texas Instruments TAC5x1x family
audio codec. These bindings define the ALSA audio interface and
regulator configuration.

Signed-off-by: Niranjan H Y <niranjan.hy@ti.com>
---
 .../devicetree/bindings/sound/ti,tac5x1x.yaml | 49 +++++++++++++++++++
 1 file changed, 49 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/sound/ti,tac5x1x.yaml

diff --git a/Documentation/devicetree/bindings/sound/ti,tac5x1x.yaml b/Documentation/devicetree/bindings/sound/ti,tac5x1x.yaml
new file mode 100644
index 000000000000..05fc93027fea
--- /dev/null
+++ b/Documentation/devicetree/bindings/sound/ti,tac5x1x.yaml
@@ -0,0 +1,49 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/sound/ti,tac5x1x.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: TI TAC5x1x Audio Codec
+
+maintainers:
+  - Niranjan H Y <niranjan.hy@ti.com>
+
+description: |
+  The TAC5x1x codec provides audio playback and capture functionality
+  with support for various audio formats and sample rates.
+
+  This binding describes the codec functionality of the TAC5x1x device,
+  which is instantiated as a child device of the main TAC5x1x MFD described
+  in Documentation/devicetree/bindings/mfd/ti,tac5x1x.yaml
+
+properties:
+  compatible:
+    const: ti,tac5x1x-codec
+
+  '#sound-dai-cells':
+    const: 0
+
+  clocks:
+    maxItems: 1
+    description: Master clock for audio processing
+
+  clock-names:
+    items:
+      - const: mclk
+
+required:
+  - compatible
+  - '#sound-dai-cells'
+
+additionalProperties: false
+
+examples:
+  - |
+    codec {
+        compatible = "ti,tac5x1x-codec";
+        #sound-dai-cells = <0>;
+        clocks = <&audio_mclk>;
+        clock-names = "mclk";
+    };
+
-- 
2.43.0

